tests/external_population_test.go

```diff
@@ -0,0 +1,195 @@
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Verifying pvc was created")
+		pvc, err := utils.WaitForPVC(f.K8sClient, dataVolume.Namespace, dataVolume.Name)
+		Expect(err).ToNot(HaveOccurred())
+		err = utils.WaitForPersistentVolumeClaimPhase(f.K8sClient, pvc.Namespace, corev1.ClaimBound, pvc.Name)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Verifying PVC's content")
+		f.ExpectEvent(dataVolume.Namespace).Should(ContainSubstring(controller.ExternalPopulationSucceeded))
+		// We add an additional newline char to replicate the populated file
+		expectetHash := []byte(expectedContent + "\n")
+		expectedHashString := fmt.Sprintf("%x", md5.Sum(expectetHash))
+		md5, err := f.GetMD5(f.Namespace, pvc, utils.DefaultPvcMountPath, 0)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(md5).To(Equal(expectedHashString))
+
+		By("Delete verifier pod")
+		err = utils.DeleteVerifierPod(f.K8sClient, f.Namespace.Name)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Cleaning up")
+		err = utils.DeleteDataVolume(f.CdiClient, f.Namespace.Name, dataVolume.Name)
+		Expect(err).ToNot(HaveOccurred())
+		Eventually(func() bool {
+			_, err := f.K8sClient.CoreV1().PersistentVolumeClaims(f.Namespace.Name).Get(context.TODO(), dataVolume.Name, metav1.GetOptions{})
+			return k8serrors.IsNotFound(err)
+		}, timeout, pollingInterval).Should(BeTrue())
+	})
+
+	It("Should not populate PVC when AnyVolumeDataSource is disabled", func() {
+		if !f.IsCSIVolumeCloneStorageClassAvailable() {
+			Skip("No CSI drivers available - Population not supported")
+		}
+		if isAnyVolumeDataSourceEnabled() {
+			Skip("AnyVolumeDataSource is enabled - Population will succeed")
+		}
+
+		By(fmt.Sprintf("Creating new datavolume %s", dataVolumeName))
+		dataVolume := utils.NewDataVolumeWithExternalPopulation(dataVolumeName, "100Mi", f.CsiCloneSCName, corev1.PersistentVolumeMode(corev1.PersistentVolumeBlock), dataSourceRef)
+		controller.AddAnnotation(dataVolume, controller.AnnDeleteAfterCompletion, "false")
+		dataVolume, err := utils.CreateDataVolumeFromDefinition(f.CdiClient, f.Namespace.Name, dataVolume)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Verifying pvc was created")
+		pvc, err := utils.WaitForPVC(f.K8sClient, dataVolume.Namespace, dataVolume.Name)
+		Expect(err).ToNot(HaveOccurred())
+		f.ForceBindIfWaitForFirstConsumer(pvc)
+		// We check the expected event
+		f.ExpectEvent(dataVolume.Namespace).Should(ContainSubstring(controller.NoAnyVolumeDataSource))
+
+		By("Cleaning up")
+		err = utils.DeleteDataVolume(f.CdiClient, f.Namespace.Name, dataVolume.Name)
+		Expect(err).ToNot(HaveOccurred())
+		Eventually(func() bool {
+			_, err := f.K8sClient.CoreV1().PersistentVolumeClaims(f.Namespace.Name).Get(context.TODO(), dataVolume.Name, metav1.GetOptions{})
+			return k8serrors.IsNotFound(err)
+		}, timeout, pollingInterval).Should(BeTrue())
+	})
+
+	It("Should not populate PVC when CSI drivers are not available", func() {
+		if utils.DefaultStorageClassCsiDriver != nil {
```

Comment:
how about checking if "local" sc exists and using that?  Otherwise, when will this be exercised?